utils/demo/helpers/createImageIdsAndCacheMetaData.js

```diff
@@ -64,13 +65,18 @@ export default async function createImageIdsAndCacheMetaData({
     );
     return imageId;
   });
+
   // if the image ids represent multiframe information, creates a new list with one image id per frame
   // if not multiframe data available, just returns the same list given
   imageIds = convertMultiframeImageIds(imageIds);
+
   imageIds.forEach((imageId) => {
     let instanceMetaData =
       cornerstoneWADOImageLoader.wadors.metaDataManager.get(imageId);
-    instanceMetaData = JSON.parse(JSON.stringify(instanceMetaData));
+
+    // It was using JSON.parse(JSON.stringify(...)) before but it is 8x slower
+    instanceMetaData = removeInvalidTags(instanceMetaData);
```

Comment:
i'm not sure why we were doing the json strigify either, do we really need this? does it really throw error if we have null or undefiend?

Reply:
Yes, it does fail when there are null/undefined values but at least it is taking 110ms vs 936ms before for the PT/CT demo dataset we are using to test it.

The only issue that I see after removing `JSON.stringify` is related to this [change](https://github.com/cornerstonejs/cornerstoneWADOImageLoader/blob/master/src/imageLoader/wadors/metaDataManager.js#L10) because it adds `isMultiFrame: undefined`. We would have to remove it from the json and keep it private to the `metadataManager` context or make sure it's set to `false` instead of `undefined`. But even after making this change I would not be 100% sure if it would work for all scenarios because `JSON.stringify` was added one year ago while that `isMultiFrame` property was added only one month ago. It may have another reason for calling it.